include/mqt-core/dd/Package.hpp

```diff
@@ -2122,6 +2071,77 @@ template <class Config> class Package {
                                        mCachedEdge::zero()});
   }
 
+  //  mEdge reduceAncillaeRecursion(mEdge& e, const std::vector<bool>&
+  //  ancillary,
+  //                                Qubit var, const bool regular = true) {
+  //
+  //    auto f = e;
+  //    std::array<mEdge, NEDGE> edges{};
+  //
+  //    // Check if ancillary at this level
+  //    if (ancillary[var]) {
+  //      // Check if level is above DD
+  //      if (f.p->v < var) {
+  //        // Create ancillaries above the DD
+  //        for (auto i = 0U; i < NEDGE; ++i) {
+  //          if (i == 0) {
+  //            edges[i] = mEdge::terminal(Complex::one());
+  //          } else {
+  //            edges[i] = mEdge::terminal(Complex::zero());
+  //          }
+  //        }
+  //        auto extension = makeDDNode(var, edges);
+  //        var = var - 1;
+  //        while (ancillary[var]) {
+  //          auto node = makeDDNode(var, edges);
+  //          extension = kronecker(extension, node, false);
+  //          var = var - 1;
+  //        }
+  //        // Stick them together
+  //        f = kronecker(extension, f, false);
+  //      } else if (f.p->v == var) {
+  //        // Replace current nodes with ancillaries
+  //        for (auto i = 0U; i < NEDGE; ++i) {
+  //          if (i == 0) {
+  //            edges[i] = {f.p->e[i].p, Complex::one()};
+  //          } else {
+  //            edges[i] = {f.p->e[i].p, Complex::zero()};
+  //          }
+  //        }
+  //        f = makeDDNode(var, edges);
+  //      } else if (f.p->v > var) {
+  //        // Create ancillaries below the DD
+  //        for (auto i = 0U; i < NEDGE; ++i) {
+  //          if (i == 0) {
+  //            edges[i] = mEdge::terminal(Complex::one());
+  //          } else {
+  //            edges[i] = mEdge::terminal(Complex::zero());
+  //          }
+  //        }
+  //        auto extension = makeDDNode(var, edges);
+  //        var = var - 1;
+  //        while (ancillary[var]) {
+  //          auto node = makeDDNode(var, edges);
+  //          extension = kronecker(extension, node, false);
+  //          var = var - 1;
+  //        }
+  //        // Stick them together
+  //        f = kronecker(f, extension, false);
+  //      }
+  //
+  //      // No ancillary
+  //    } else {
+  //      for (auto i = 0U; i < NEDGE; ++i) {
+  //        edges[i] =
+  //            reduceAncillaeRecursion(f.p->e[i], ancillary, var - 1, regular);
+  //      }
+  //      f = makeDDNode(var, edges);
+  //    }
+  //
+  //    f.w = cn.lookup(f.w * e.w);
+  //    return f;
+  //  }
+
   vCachedEdge reduceGarbageRecursion(vNode* p, const std::vector<bool>& garbage,
```

Comment:
GitHub won;t let me comment on the right line, but the 
```c++
mCachedEdge reduceGarbageRecursion(mNode* p, const std::vector<bool>& garbage,
                                     const Qubit lowerbound,
                                     const bool regular = true)
```
method also needs adaptation to the new structure.